src/modbus/ModbusClient.cpp

```diff
@@ -27,11 +28,14 @@ ModbusClient::ModbusClient(std::chrono::milliseconds responseTimeout)
 
 bool ModbusClient::connect()
 {
+    m_connected = false;
+
     std::lock_guard<decltype(m_modbusMutex)> l{m_modbusMutex};
-    if (!createContext())
+    if (!m_contextCreated && !createContext())
     {
         return false;
     }
+    m_contextCreated = true;
 
     if (modbus_set_response_timeout(m_modbus, 2, 0) == -1)
```

Comment:
Destroying context below will leave in an inconsistent state with both m_contextCreated and m_connected  set to true